src/pepr/operator/controllers/keycloak/client-sync.ts

```diff
@@ -142,6 +144,18 @@ async function syncClient(
   }
 }
 
+/**
+ * Handles the client groups by converting the groups to attributes.
+ * @param clientReq - The client request object.
+ */
+function handleClientGroups(clientReq: Sso) {
+  if (clientReq.groups) {
+    clientReq.attributes = clientReq.attributes || {};
+    clientReq.attributes["uds.core.groups"] = JSON.stringify(clientReq.groups);
+    delete clientReq.groups;
+  }
+}
```

Comment:
I would still like to see us put an actual allowlist on the allowed keys for `attributes`. Technically that can go in a separate PR, but if we don't do it now please create an issue.